src/main/java/de/thomaskrille/dropwizard_template_config/TemplateConfigurationSourceProvider.java

```diff
@@ -69,8 +75,15 @@ public InputStream open(final String path) throws IOException {
 
             new Template("config", configTemplate, configuration)
                     .process(dataModel, new OutputStreamWriter(processedTemplateStream, charset));
+            byte[] processedTemplateBytes = processedTemplateStream.toByteArray();
 
-            return new ByteArrayInputStream(processedTemplateStream.toByteArray());
+            if (outputPath.isPresent()) {
+                File outputFile = new File(outputPath.get());
+                Files.createParentDirs(outputFile);
+                Files.write(processedTemplateBytes, outputFile);
```

Comment:
If the user has configured the package to write out the config, it might be reasonable not to handle exceptions here and just let the exception abort startup.